knowledge_repo/app/templates/index-feed.html

```diff
@@ -1,81 +1,72 @@
 {% extends "index-base.html" %}
 
-{% block inner_content %}
+{% block style_links %}
+{{ super() }}
+<link rel="stylesheet" href="{{ url_for('static', filename='css/pages/index.css')}}">
+{% endblock %}
+
+{% macro render_tag(tag, post_id) %}
```

Comment:
The macro just made it easier for me to read the html that renders the post. Apart from that, there was a mild interest in separating form from function. Most of the structure of the post rows are now semantic, whereas the html for the tags was a little more involves a little more structure. If you prefer it all inline, happy to rearrange it.